### PR TITLE
feat: add dateonly converter

### PR DESCRIPTION
**Pull request by danield20**
Adding datateonly and timeonly convertors (taken from .net 8) to coreWF

https://uipath.atlassian.net/browse/SW-14266

### CODE REVIEW

File: src/System.Xaml/System/Windows/Markup/TypeConverterHelper.cs [hunk 1]
```diff
@@ -244,6 +244,10 @@ internal static Type GetCoreConverterTypeFromCustomType(Type type)
             {
                 converterType = typeof(DateTimeConverter2);
             }
+            else if (typeof(DateOnly).IsAssignableFrom(type))
```
Comment (mirceaarhip) [inline src/System.Xaml/System/Windows/Markup/TypeConverterHelper.cs:247]: what about TimeOnly

File: src/System.Xaml/System/Xaml/Replacements/DateOnlyConvertor.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+using System.ComponentModel.Design.Serialization;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert <see cref='System.DateOnly'/> objects to and from various other representations.
+    /// </summary>
+    public class DateOnlyConverter : TypeConverter
```
Comment (mirceaarhip) [inline src/System.Xaml/System/Xaml/Replacements/DateOnlyConvertor.cs]: let's use a different name so we won't  have conflicts with the real type

Comment (mirceaarhip) [reply src/System.Xaml/System/Xaml/Replacements/DateOnlyConvertor.cs]: an the namespace

**Review by mirceaarhip — CHANGES_REQUESTED** [review]

File: src/System.Xaml/System/Xaml/Replacements/TimeOnlyConverter2.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+using System.ComponentModel.Design.Serialization;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace System.Xaml.Replacements
+{
+    /// <summary>
+    /// Provides a type converter to convert <see cref='System.TimeOnly'/> objects to and from various other representations.
+    /// </summary>
+    public class TimeOnlyConverter2 : TypeConverter
```
Comment (mirceaarhip) [inline src/System.Xaml/System/Xaml/Replacements/TimeOnlyConverter2.cs]: can this be internal? Also not convinced this is a good name.

**Review by mirceaarhip — CHANGES_REQUESTED** [review]

File: src/System.Xaml/System/Xaml/Replacements/TimeOnlyConverter2.cs (hunk 3 above)
Comment (danield20) [reply src/System.Xaml/System/Xaml/Replacements/TimeOnlyConverter2.cs]: changed them to internal. The name is similar to the ones previously used eg. `DateTimeConverter2`

**Review by mirceaarhip — APPROVED** [review]

**Review by aoltean16 — APPROVED** [review]

**Pull request merged**